llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1502,6 +1542,16 @@ bool NVPTXDAGToDAGISel::tryStoreVector(SDNode *N) {
     N2 = N->getOperand(5);
     ToTypeWidth = TotalWidth / 4;
     break;
+  case NVPTXISD::StoreV8:
+    if (!Subtarget->has256BitMaskedLoadStore())
+      return false;
```

Comment:
Should this be an assert? Is there a valid case where this would occur?

Reply:
Yes, you're right, it should be an assert. Type legalization shouldn't let anything past that doesn't pass that check.

Reply:
I'm actually probably going to drop this check all together, I think it's not needed, and if we wanted to include it, we should also be checking that API for 4 x 64.

Reply:
Works for me